### PR TITLE
2.0 framework update

### PR DESCRIPTION
**Pull request by fhamon**
🔥

### CODE REVIEW

File: src/utils/keys.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		});
+		return key;
+	};
+
+	// Chars
+	global.isChar = function (c) {
+		return c === window.keys.space_bar || (c > window.keys['0'] && c <= window.keys.z);
+	};
+
+})(jQuery, window);
```
Comment (nitriques) [inline src/utils/keys.js]: 1. Please add an empty line at the end.
1. Please put the keys into `App`

Comment (fhamon) [reply src/utils/keys.js]: @nitriques How about App. **extensions** .keys? I don't think code that does not contribute directly to the framework pollute the App namespace.

Comment (fhamon) [reply src/utils/keys.js]: @nitriques check 385691d81b8ffab1a7a5eef6863bec692eaba885

**Comment by fhamon** [comment]
@nitriques ping !

**Comment by nitriques** [comment]
Yup I'll merge as soon as fx 2 is out.

**Comment by nitriques** [comment]
Can you please rebase ?

**Comment by nitriques** [comment]
@fhamon game to rebase ?

**Comment by nitriques** [comment]
@fhamon We need a rebase here, sorry

**Comment by nitriques** [comment]
@fhamon :)

**Comment by fhamon** [comment]
wilco

**Comment by nitriques** [comment]
@fhamon  Soon please, this is blocking the whole release !

**Comment by fhamon** [comment]
Sorry in a rush. will do at the end of this week or next week

File: src/utils/keys.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	};
+
+	// Chars
+	keys.isChar = function (c) {
+		return c === App.device.keys.code.space_bar ||
+			(c > App.device.keys.code['0'] && c <= App.device.keys.code.z);
+	};
+
+	window.App = $.extend(true, window.App, {
+		keys: keys
```
Comment (nitriques) [inline src/utils/keys.js]:
```suggestion
		device: {
			keys: keys
		}
```

File: src/utils/keys.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+			}
+
+			return true;
+		});
+		return key;
+	};
+
+	// Chars
+	keys.isChar = function (c) {
+		return c === App.device.keys.code.space_bar ||
```
Comment (nitriques) [inline src/utils/keys.js]:
```suggestion
		return c === keys.code.space_bar ||
```

File: src/utils/keys.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+			(c > App.device.keys.code['0'] && c <= App.device.keys.code.z);
```
Comment (nitriques) [inline src/utils/keys.js]:
```suggestion
			(c > keys.code['0'] && c <= keys.code.z);
```

**Comment by nitriques** [comment]
The build is also broken.

We can ignore all the values inside the big ass object @fhamon

**Comment by fhamon** [comment]
@nitriques done.

File: src/utils/keys.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+ * Friendly maping of "all" the keyboard keys
+ * @author Deux Huit Huit
+ */
+(function ($) {
+
+	'use strict';
+
+	var keys = {};
+	
+	/* jshint ignore:start */
```
Comment (nitriques) [inline src/utils/keys.js]: This needs to after line 12

File: src/utils/keys.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+	keys.code = {
```
Comment (nitriques) [inline src/utils/keys.js]:
```suggestion
	keys.code = {
		/* jshint ignore:start */
```

File: src/utils/keys.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+		forward_slash: 191,
+		'/': 191,
+		grave_accent: 192,
+		open_bracket: 219,
+		back_slash: 220,
+		'\\': 220,
+		close_braket: 221,
+		single_quote: 222
+	};
+	/* jshint ignore:end */
```
Comment (nitriques) [inline src/utils/keys.js]: remove here

File: src/utils/keys.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+		period: 190,
+		forward_slash: 191,
+		'/': 191,
+		grave_accent: 192,
+		open_bracket: 219,
+		back_slash: 220,
+		'\\': 220,
+		close_braket: 221,
+		single_quote: 222
+	};
```
Comment (nitriques) [inline src/utils/keys.js]:
```suggestion
	/* jshint ignore:end */
	};
```

**Pull request merged**